client/txn.go

```diff
@@ -76,6 +76,7 @@ type Txn struct {
 	// is (still) unset.
 	haveTxnWrite bool
 	haveEndTxn   bool // True if there was an explicit EndTransaction
+	execing      bool // True if the txn is run in the exec() loop
```

Comment:
instead of having this flag, how about separating `Commit()` into `Commit()` and `commit()`? `Commit()` would include the cleanup logic and the `exec` loop would call `commit()`.